### PR TITLE
Include ELF requirements and provides in dist-info

### PR DESCRIPTION
**Pull request by tiran**
The `add_extra_metadata_to_wheels()` step now writes
`fromager-elf-provides.txt` and `fromager-elf-requires.txt` files with
provided and required shared object names.

I have also fixed several minor issues with the extraction code. It now
analyzes any file with a Python extension suffix and any files that
matches `lib*.so*` pattern.

Fixes: #338

### CODE REVIEW

**Comment by tiran** [comment]
Depends on #344

**Comment by shubhbapna** [comment]
I think it needs a rebase? #344 is showing up as changes in the diff

**Comment by tiran** [comment]
The changes in this PR conflicted with changes from the build tag PR. I have addresses the conflict.

File: src/fromager/wheels.py [hunk 1]
```diff
@@ -81,31 +85,65 @@ def _createenv(self) -> None:
...
+    settings: elfdeps.ELFAnalyzeSettings,
+) -> typing.Generator[elfdeps.ELFInfo, None, None]:
+    """Locate shared libraries and yield ELF information"""
+    for root, _, files in os.walk(wheel_root_dir):
+        rootpath = pathlib.Path(root)
+        for filename in files:
+            # any Python extension module or 'lib*.so*'
+            if filename.endswith(tuple(EXTENSION_SUFFIXES)) or (
+                filename.startswith("lib") and ".so" in filename
+            ):
```
Comment (shubhbapna) [inline src/fromager/wheels.py:100]: will we need this after https://github.com/python-wheel-build/elfdeps/pull/6 ?

Comment (tiran) [reply src/fromager/wheels.py]: No, we don't need this any more. I have changed the code to use the new `analyze_dirtree` function.

**Review by shubhbapna — APPROVED** [review]

**Pull request merged**